File: i2f-springboot/i2f-springboot-secure-starter/src/main/java/i2f/springboot/secure/consts/SecureErrorCode.java
package i2f.springboot.secure.consts;

/**
 * @author Ice2Faith
 * @date 2023/6/13 10:19
 * @desc
 */
public enum SecureErrorCode {
    SECURE_HEADER_EMPTY(101),
    SECURE_HEADER_STRUCTURE(102),
    SECURE_HEADER_SIGN_EMPTY(103),
    SECURE_HEADER_NONCE_EMPTY(104),
    SECURE_HEADER_RANDOM_KEY_EMPTY(105),
    SECURE_HEADER_ASYM_SIGN_EMPTY(106),
    SECURE_HEADER_DIGITAL_EMPTY(107),

    BAD_SIGN(201),
    BAD_RANDOM_KEY(202),
    BAD_NONCE(203),
    BAD_DIGITAL(204),

    BAD_SECURE_REQUEST(301);

    private int code;

    SecureErrorCode(int code) {
        this.code = code;
    }

    public int code() {
        return this.code;
    }
}

File: i2f-springboot/test-swl-starter/web/secure-web/src/secure/consts/SecureErrorCode.js
/**
 * 定义异常的错误码
 */
const SecureErrorCode = {
    SECURE_NOT_SUPPORT: () => 10,

    SECURE_HEADER_EMPTY: () => 101,
    SECURE_HEADER_STRUCTURE: () => 102,
    SECURE_HEADER_SIGN_EMPTY: () => 103,
    SECURE_HEADER_NONCE_EMPTY: () => 104,
    SECURE_HEADER_RANDOM_KEY_EMPTY: () => 105,
    SECURE_HEADER_ASYM_SIGN_EMPTY: () => 106,
    SECURE_HEADER_DIGITAL_EMPTY: () => 107,

    BAD_SIGN: () => 201,
    BAD_RANDOM_KEY: () => 202,
    BAD_NONCE: () => 203,
    BAD_DIGITAL: () => 204,

    BAD_SECURE_REQUEST: () => 301
}
export default SecureErrorCode
